app/services/authService.js

```diff
@@ -78,6 +84,35 @@ class AuthService {
     return loginInfoObj;
   }
 
+  async resetPassword(newPassword, token) {
+    // reset password
+    const tokenDoc = await PortalTokenModel.findOne({
+      token: token
+    });
+    // If token not exist
+    if (!tokenDoc) {
+      throw new CustomError(STATUS_TYPE.PORTAL_ACTIVATE_CODE_ILLEGAL);
+    }
+
+    if ((+new Date() - tokenDoc.last_access_time) / 1000 > config.tokenTimeOut) {
+      await PortalTokenModel.deleteOne({ _id: tokenDoc._id });
+      throw new CustomError(STATUS_TYPE.PORTAL_ACTIVATE_CODE_EXPIRED);
+    }
+
+    // update password 
+    const user = await PortalUserModel.findById(tokenDoc.user_id)
```

Comment:
findById(_id)
However, user_id is a param. Therefore, a different function is needed.